### PR TITLE
swap arbitrator for arbitrableProxy

### PR DESCRIPTION
**Pull request by greenlucid**
Instead of interfacing with an ERC-792 Arbitrator, this patch connects to an "ArbitrableProxy" contract.
https://github.com/kleros/arbitrable-proxy-contracts/

This contract allows to delegate common Arbitrable tasks (evidence submission and appeal crowdfunding logic) to a different contract on its behalf

I hope that, even if it's not what you intended, the logic is simple to follow and you can adapt it to your needs.

## Main Changes

1. Remove ERC-792 and ERC-1497 direct interfacing from `userregistry.sol` and `nftprotect.sol`
2. Remove appeals, and evidence submissions
3. Remove appeal based events
4. Make the ArbitratorRegistry register ArbitrableProxies instead.

### Why ArbitratorRegistry?

Why is the Arbitrator Registry available? If allowed to choose, attackers will always choose the cheapest or most vulnerable arbitrator available. Why provide the attackers with that choice? In `nftprotect.sol`, it appears creators of requests can choose whatever Arbitrator (now, ArbitrableProxy) they want

### Initial Evidence Submission

Since NFTProtect is no longer interfacing with ERC-1497, it is no longer possible to emit evidence internally. Regular evidence (the one emitted by the `Evidence(...)` event) with show when the emitter contract is the Arbitrable. In this case, the _real_ Arbitrable will be the ArbitrableProxy, not NFTProtect.

I put the evidence as a field in the events that signal the creation of events. That way, they could be highlighted and rendered as part of the Evidence Display provided in the metaevidence for those requests, or in the frontend. That way they will also be distinct from normal Evidence (since, they were the pieces of evidence that signal why the request is needed in the first place)

Some requests method didn't even require this, since you can call `arbitrableProxy.submitEvidence(...)` and make the proxy emit an Evidence event. But, you can only reliably do this when you know the disputeId of a request, and some requests didn't create disputes as soon as they were launched.

### Requests don't necessarily entail Disputes

Previously, in order to conform to ERC-792, `nftprotect.sol` needed to keep track of the `extraData` to pass it to the appeal method later. This is no longer the case.
However, there are some requests that do not launch disputes immediately upon creation. These are the ones that are created through `askOwnershipAdjustment`. This enforces the Request struct to store the `disputeId`, and keep mappings between disputeId => request.

This complexity, logic and storage overhead could be removed by removing that possibility.

### Need of storing Arbitrator

Previously, in order to call the appeals, you needed to store the `arbitrator` address. This is no longer the case, as appeals are no longer called from NFTProtect.
However, you need to store the "arbitrator" (in this case, the Arbitrable Proxy) in order to fetch the ruling.

You cannot simply store an id to an entry in the ArbitratorRegistry (to remove the need of storing many fields separatedly), because it's mutable. So, the governor of this registry could call `deleteArbitrator(...)` and then you could no longer fetch the ruling later.

### Disputes from Requests that didn't immediately launch, could launch with latest available parameters

Instead of needing to store the `extraData` and the `metaEvidence`, you could simply use the latest available parameters for these. That way you could remove those properties from the `Request` struct.

### CODE REVIEW

**Review by odubinkin — APPROVED** [review]

**Comment by odubinkin** [comment]
Hi @greenlucid, thanks for your PR!
All the changes you made looks clear for me and actually I think it's a good idea to move all the arbitration logic out of the main contract. Especially since the size of the nftprotect contract has already reached the limit of 24Kb.
I'll wait for a review from @isvirin before merging it.
I'll try to answer your observations below.

**Comment by odubinkin** [comment]
- Why ArbitratorRegistry?
We are considering supporting more arbitrators in the future. Since this will initially be a list of arbitrators approved by us (or later by the DAO), we assume that they all have equal competence to resolve cases. Therefore, it is not an attack vector.

- Requests don't necessarily entail Disputes
Creating a request on our side does not always involve opening a dispute. For example, confirming the transfer of the original NFT would only require opening a dispute if the ultra (paranoid) protection is being used. And while we do not open a dispute, it is also not required to provide evidence URI.  One requestId can correspond to from 0 to 1 disputeId, one disputeId always corresponds to 1 requestId.

- Need of storing Arbitrator
Great observation about the deleteArbitrator method. @isvirin and I will discuss what to do about it.

- Disputes from Requests that didn't immediately launch, could launch with latest available parameters
You're right, looks like metaEvidence and extraData are not needed inside a Request struct. @isvirin, are we missing anything?

**Comment by isvirin** [comment]
Hi @greenlucid @odubinkin 
I find it's very good idea to create an ArbitrableProxy abstraction. Saying about "Observations", sure we can simplify contract, but not earlier, than we simplify business logic of our app. We have many ideas to implement after MVP and I'll keep in mind all there points.
@odubinkin please merge this PR and I will work on reducing contract size to make it possible to deploy to mainnet.

**Pull request merged**
